treeherder/model/management/commands/backfill_text_log_error_jobs.py

```diff
@@ -0,0 +1,46 @@
+import logging
+from django.core.management.base import BaseCommand
+
+from treeherder.model.models import TextLogError
+from treeherder.utils.queryset import chunked_qs
+
+
+logger = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    """Management command to manually initiate intermittent failures commenter
+       default is daily and non-test mode, use flags for weekly, auto and test mode."""
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--chunk-size',
+            action='store',
+            dest='chunk_size',
+            default=1000,
+            type=int,
+            help=('Define the size of the chunks for querying the TextLogError table'),
+        )
+
+    def handle(self, *args, **options):
+        queryset = TextLogError.objects.select_related('step').filter(job__isnull=True)
+        chunk_size = options['chunk_size']
+
+        for chunked_queryset in chunked_qs(
+            queryset, chunk_size=chunk_size, fields=['id', 'step', 'job']
+        ):
+            if not chunked_queryset:
+                return
+
+            for row in chunked_queryset:
+                row.job_id = row.step.job_id
```

Comment:
Are you able to simultaneously remove the FK to step here?  Or does that have to happen in a separate backfill?

Reply:
We need to keep it for now because we're still accessing errors via the TextLogStep API (and adding "steps") until I merge part 2. Part of what make for such a huge pr in my original one was how interconnected this all is.

Reply:
Oh right, of course.